### PR TITLE
SC-7108 - create a component header

### PR DESCRIPTION
**Pull request by LukasSielski**
# Description
Header for the course details page.

## Links to Tickets or other pull requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-7108

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by LukasSielski** [comment]
@schul-cloud-bot update screenshots

**Comment by LukasSielski** [comment]
@schul-cloud-bot update screenshots

**Review by haniapalac — CHANGES_REQUESTED** [review]

File: src/components/base/BaseHeader.vue [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		},
+	},
+	data() {
+		return {
+			active: false,
+		};
+	},
+	computed: {
+		hasActions() {
+			return this.$attrs.hasOwnProperty("actions");
```
Comment (pkleybolte) [inline src/components/base/BaseHeader.vue:42]: I would prefer to add `actions` to `props`. It is way easier to see what attribute this component needs that way.

File: src/components/base/BaseHeader.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<template>
+	<div>
+		<div class="header-container">
+			<div class="header-top">
+				<span class="header-title">{{ title }}</span>
+				<span v-if="hasActions" class="ctx-menu">
+					<base-button design="text icon" @click="triggerContextMenu">
+						<base-icon v-bind="$attrs" />
+					</base-button>
+					<context-menu :show.sync="active" v-bind="$attrs" v-on="$listeners" />
```
Comment (pkleybolte) [inline src/components/base/BaseHeader.vue:10]: `v-bind="$attrs"` could be written as `:actions="actions"`, right?
Unfortunately I don't see a nicer solution for `v-on="$listeners"` though.

File: src/components/molecules/CourseHeader.unit.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import CourseHeader from "./CourseHeader";
+
+describe("@components/molecules/CourseHeader", () => {
+	it(...isValidComponent(CourseHeader));
```
Comment (pkleybolte) [inline src/components/molecules/CourseHeader.unit.js:4]: Please add tests:
it("should show nextLessonDate",...)
it("renders a link with correct url",...)

**Review by pkleybolte — CHANGES_REQUESTED** [review]

File: src/components/base/BaseHeader.vue [hunk 4]
```diff
@@ -0,0 +1,70 @@
+<template>
+	<div>
+		<div class="header-container">
+			<div class="header-top">
+				<span class="header-title">{{ title }}</span>
+				<span v-if="hasActions" class="ctx-menu">
+					<base-button design="text icon" @click="triggerContextMenu">
```
Comment (ABialoskorski) [inline src/components/base/BaseHeader.vue:7]: I would do @click="this.active = !this.active" so You don't need method for that

**Comment by LukasSielski** [comment]
@schul-cloud-bot update screenshots

**Review by haniapalac — APPROVED** [review]

**Comment by LukasSielski** [comment]
Please don't merge it yet as some UX addjustments needs to be done.

**Comment by LukasSielski** [comment]
@schul-cloud-bot update screenshots

**Review by pkleybolte — APPROVED** [review]

**Review by haniapalac — APPROVED** [review]

**Pull request merged**